### PR TITLE
fix: correct opt-out logic

### PR DESCRIPTION
**Pull request by scottschreckengaust**
Update enableOperationalMetric value when set to `null` or `undefined`.

Also, adding to the logic the version and type of class along with the construct's identifier.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Review by krokoko — DISMISSED** [review]

**Review by hvital — APPROVED** [review]

File: src/patterns/gen-ai/aws-rag-appsync-stepfn-opensearch/index.ts [hunk 1]
```diff
@@ -693,8 +693,9 @@ export class RagAppsyncStepfnOpensearch extends Construct {
     );
 
 
-    const enableOperationalMetric = props.enableOperationalMetric || true;
-    const solution_id = 'genai_cdk_'+id;
+    const enableOperationalMetric =
+      props.enableOperationalMetric !== undefined && props.enableOperationalMetric !== null ? props.enableOperationalMetric : true;
+    const solution_id = 'genai_cdk_' + version + '/' + typeof(this) + '/' + id;
```
Comment (laithalsaadoon) [inline src/patterns/gen-ai/aws-rag-appsync-stepfn-opensearch/index.ts]: Consider template literal syntax: ```const solution_id = `genai_cdk_${version}/${typeof(this)}/${id}` ```

File: src/patterns/gen-ai/aws-summarization-appsync-stepfn/index.ts [hunk 2]
```diff
@@ -705,8 +705,9 @@ export class SummarizationAppsyncStepfn extends Construct {
     this.processedAssetBucket?.grantReadWrite(documentReaderLambda);
 
 
-    const enableOperationalMetric = props.enableOperationalMetric || true;
-    const solution_id = 'genai_cdk_'+id;
+    const enableOperationalMetric =
+      props.enableOperationalMetric !== undefined && props.enableOperationalMetric !== null ? props.enableOperationalMetric : true;
+    const solution_id = 'genai_cdk_' + version + '/' + typeof(this) + '/' + id;
```
Comment (laithalsaadoon) [inline src/patterns/gen-ai/aws-summarization-appsync-stepfn/index.ts]: `solution_id` should be camel case

File: src/patterns/gen-ai/aws-summarization-appsync-stepfn/index.ts [hunk 3]
```diff
@@ -705,8 +705,9 @@ export class SummarizationAppsyncStepfn extends Construct {
...
 
     if (enableOperationalMetric) {
       documentReaderLambda.addEnvironment(
```
Comment (laithalsaadoon) [inline src/patterns/gen-ai/aws-summarization-appsync-stepfn/index.ts:713]: consider moving solutionId instantiation behind the conditional if solutionId isn't used elsewhere

**Review by laithalsaadoon — DISMISSED** [review]
some idiomatic/Node-ism nitpicks

**Review by krokoko — APPROVED** [review]

**Review by laithalsaadoon — COMMENTED** [review]
👍

**Review by dineshSajwan — APPROVED** [review]
LGTM

**Pull request merged**